staging/src/kubevirt.io/api/export/v1alpha1/types.go

```diff
@@ -54,8 +54,9 @@ type VirtualMachineExportList struct {
 type VirtualMachineExportSpec struct {
 	Source corev1.TypedLocalObjectReference `json:"source"`
 
-	// TokenSecretRef is the name of the secret that contains the token used by the export server pod
-	TokenSecretRef string `json:"tokenSecretRef"`
+	// +optional
+	// TokenSecretRef is the name of the custom-defined secret that contains the token used by the export server pod
+	TokenSecretRef string `json:"tokenSecretRef,omitempty"`
```

Comment:
Out of curiosity, I'm seeing that we use regular strings for all other fields in the export API (even in the Spec's `TokenSecretRef`). Why we'd want this one to be a pointer?

Reply:
I have seen CSI APIs use `type LocalObjectReference struct ` for `SecretRef` but then on the other hand we have CDI's which are strings (`type DataVolumeSourceRegistry struct` for example)

Reply:
> Out of curiosity, I'm seeing that we use regular strings for all other fields in the export API (even in the Spec's TokenSecretRef). Why we'd want this one to be a pointer?

we are specifically talking about the `spec.tokenSecretRef`

Since this field is optional, it should be a pointer per kubernetes api conventions.  https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Of course we are not always consistent about this